lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -202,7 +209,12 @@ struct ReduceOpConversion
         accIndex = accIndices[key];
       SmallVector<Value> writeIdx = indices[key];
 
-      writeIdx[axis] = udiv(writeIdx[axis], sizePerThread);
+      if (srcLayout.isa<BlockedEncodingAttr>()) {
+        writeIdx[axis] = udiv(writeIdx[axis], axisSizePerThread);
+      } else {
+        writeIdx[axis] =
+            add(mul(udiv(writeIdx[axis], _16), _8), urem(writeIdx[axis], _8));
```

Comment:
Does that work for mmav1?

Reply:
Also a comment that describes why this line is necessary would help

Reply:
I'll add support for mmav1 as well